### PR TITLE
M3C 10 cards

### LINKED ISSUE #5359: Incorrect payment cost with Tarmogoyf Nest
**Describe the bug**
Currently the cost to create a Tarmogoyf token using *Tarmogoyf Nest* is {T}, but it should be {1}{G}, {T}.

**Desktop (please complete the following information):**
 - Forge Version: forge-gui-desktop-1.6.63-SNAPSHOT-06-01

### PR DESCRIPTION
**Pull request by Simisays**
Closes #5359

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/sunken_palace.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Name:Sunken Palace
+ManaCost:no cost
+Types:Land Cave
+K:CARDNAME enters the battlefield tapped.
+A:AB$ Mana | Cost$ T | Produced$ U | SpellDescription$ Add {U}.
+A:AB$ Mana | Cost$ 1 U T ExileFromGrave<5/Card>| Produced$ U | TriggersWhenSpent$ TrigCopy | SpellDescription$ Add {U}. When you spend this mana to cast a spell or activate an ability, copy that spell or ability. You may choose new targets for the copy. (Mana abilities can't be copied.)
+SVar:TrigCopy:Mode$ SpellAbilityCast | ValidSA$ Spell,Activated | ValidActivatingPlayer$ You | Execute$ TrigCopyMain | TriggerDescription$ When you spend this mana to cast a spell or activate an ability, copy that spell or ability. You may choose new targets for the copy. (Mana abilities can’t be copied.)
```
Comment (Simisays) [inline forge-gui/res/cardsfolder/upcoming/sunken_palace.txt:7]: This is not working

File: forge-gui/res/cardsfolder/upcoming/desert_warfare.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Name:Desert Warfare
+ManaCost:3 G
+Types:Enchantment
+T:Mode$ Sacrificed | ValidPlayer$ You | ValidCard$ Desert.YouCtrl | Execute$ TrigDelay | TriggerZones$ Battlefield | TriggerDescription$ Whenever you sacrifice a Desert and whenever a Desert card is put into your graveyard from your hand or library, put that card onto the battlefield under your control at the beginning of your next end step.
+T:Mode$ ChangesZone | ValidCard$ Desert.YouOwn | Origin$ Hand,Library | Destination$ Graveyard | TriggerZones$ Battlefield | Execute$ TrigDelay | Secondary$ True | TriggerDescription$ Whenever you sacrifice a Desert and whenever a Desert card is put into your graveyard from your hand or library, put that card onto the battlefield under your control at the beginning of your next end step.
+SVar:TrigDelay:DB$ DelayedTrigger | Mode$ Phase | RememberObjects$ TriggeredCard  | Phase$ End of Turn | ValidPlayer$ Player | Execute$ TrigReturn | TriggerDescription$ Put that card onto the battlefield under your control at the beginning of your next end step.
+SVar:TrigReturn:DB$ ChangeZone | Defined$ DelayTriggerRememberedLKI | Origin$ Graveyard | Destination$ Battlefield
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/desert_warfare.txt:7]: "your control"

File: forge-gui/res/cardsfolder/upcoming/desert_warfare.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+T:Mode$ Phase | IsPresent$ Desert.YouCtrl | PresentCompare$ GE5 | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigToken | TriggerZones$ Battlefield | TriggerDescription$ At the beginning of combat on your turn, if you control five or more Deserts, create that many 1/1 red, green, and white Sand Warrior creature tokens. They gain haste.
+SVar:TrigToken:DB$ Token | TokenAmount$ X | PumpKeywords$ Haste | TokenScript$ rgw_1_1_sand_warrior | TokenOwner$ You
+SVar:X:Count$Valid Desert.YouCtrl
+DeckNeeds:Type$Desert
+DeckHas:Ability$Token & Type$Sand
+Oracle:Whenever you sacrifice a Desert and whenever a Desert card is put into your graveyard from your hand or library, put that card onto the battlefield under your control at the beginning of your next end step.\nAt the beginning of combat on your turn, if you control five or more Deserts, create that many 1/1 red, green, and white Sand Warrior creature tokens. They gain haste
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/desert_warfare.txt:13]: .

File: forge-gui/res/cardsfolder/upcoming/rampant_frogantua.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Name:Rampant Frogantua
+ManaCost:2 G
+Types:Creature Frog
+PT:3/3
+K:Trample
+S:Mode$ Continuous | AddPower$ X | AddToughness$ X | Affected$ Card.Self | Description$ CARDNAME gets +10/+10 for each player who has lost the game.
+SVar:X:PlayerCountHasLost$Amount/Times.10
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigMill | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME deals combat damage to a player, you may mill that many cards. Put any number of land cards from among them onto the battlefield tapped.
+SVar:TrigMill:DB$ Mill | Defined$ You | NumCards$ Y | RememberMilled$ True | SubAbility$ DBChangeZone
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/rampant_frogantua.txt:9]: optional missing

File: forge-gui/res/cardsfolder/upcoming/rampant_frogantua.txt [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+SVar:DBChangeZone:DB$ ChangeZone | Origin$ Graveyard | Destination$ Battlefield | ChangeType$ Land.IsRemembered | Hidden$ True | ChangeNum$ Z | AnyNumber$ True | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/rampant_frogantua.txt:10]: add exile

Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/rampant_frogantua.txt:10]: tapped missing

File: forge-gui/res/cardsfolder/upcoming/satya_aetherflux_genius.txt [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+PT:3/5
+K:Menace
+K:Haste
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigCopy | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME attacks, create a tapped and attacking token that's a copy of up to one other target nontoken creature you control. You get {E}{E} (two energy counters). At the beginning of the next end step, sacrifice that token unless you pay an amount of {E} equal to its mana value.
+SVar:TrigCopy:DB$ CopyPermanent | ValidTgts$ Creature.nonToken+YouCtrl | TgtPrompt$ Select up to one other target nontoken you control | TargetMin$ 0 | TargetMax$ 1 |  TokenTapped$ True | TokenAttacking$ True | RememberTokens$ True | SubAbility$ DBEnergy
+SVar:DBEnergy:DB$ PutCounter | Defined$ You | CounterType$ ENERGY | CounterNum$ 2 | SubAbility$ DBDelayTrig
+SVar:DBDelayTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End Of Turn | Execute$ TrigSac | TriggerDescription$ At the beginning of the next end step, sacrifice that token unless you pay an amount of {E} equal to its mana value.
+SVar:TrigSac:DB$ SacrificeAll | Defined$ Remembered | Controller$ You | UnlessCost$ PayEnergy<X> | UnlessPayer$ You | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/satya_aetherflux_genius.txt:11]: cleanup / remembering done wrong

File: forge-gui/res/cardsfolder/upcoming/sunken_palace.txt [hunk 7]
```diff
@@ -0,0 +1,9 @@
+Name:Sunken Palace
+ManaCost:no cost
+Types:Land Cave
+K:CARDNAME enters the battlefield tapped.
+A:AB$ Mana | Cost$ T | Produced$ U | SpellDescription$ Add {U}.
+A:AB$ Mana | Cost$ 1 U T ExileFromGrave<5/Card>| Produced$ U | TriggersWhenSpent$ TrigCopy | SpellDescription$ Add {U}. When you spend this mana to cast a spell or activate an ability, copy that spell or ability. You may choose new targets for the copy. (Mana abilities can't be copied.)
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/sunken_palace.txt:6]: 5 isn't 7

File: forge-gui/res/cardsfolder/upcoming/sunken_palace.txt [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+SVar:TrigCopy:Mode$ SpellAbilityCast | ValidSA$ Spell,SpellAbility.nonManaAbility | ValidActivatingPlayer$ You | Execute$ TrigCopyMain | TriggerDescription$ When you spend this mana to cast a spell or activate an ability, copy that spell or ability. You may choose new targets for the copy. (Mana abilities can’t be copied.)
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/sunken_palace.txt:7]: wrong valid

File: forge-gui/res/cardsfolder/upcoming/sunken_palace.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/sunken_palace.txt]: ?

File: forge-gui/res/cardsfolder/upcoming/sunken_palace.txt (hunk 8 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/sunken_palace.txt:7]: Any ideas ?

File: forge-gui/res/cardsfolder/upcoming/rampant_frogantua.txt (hunk 4 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/rampant_frogantua.txt:9]: that's not how it works...

File: forge-gui/res/cardsfolder/upcoming/sunken_palace.txt (hunk 8 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/sunken_palace.txt]: what was wrong with Activated?

File: forge-gui/res/cardsfolder/upcoming/satya_aetherflux_genius.txt (hunk 6 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/satya_aetherflux_genius.txt]: somewhat better but why did you switch to remembering the target...?

File: forge-gui/res/cardsfolder/upcoming/aurora_shifter.txt [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+PT:1/3
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigEnergy | TriggerDescription$ Whenever CARDNAME deals combat damage to a player, you get that many {E}.
+SVar:TrigEnergy:DB$ PutCounter | Defined$ You | CounterType$ ENERGY | CounterNum$ TriggerCount$DamageAmount
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigImmediateTrig | TriggerDescription$ At the beginning of combat on your turn, you may pay {E}{E}. When you do, CARDNAME becomes a copy of another target creature you control, except it has this ability and "Whenever this creature deals combat damage to a player, you get that many {E}."
+SVar:TrigImmediateTrig:AB$ ImmediateTrigger | Cost$ PayEnergy<2> | Execute$ TrigCopy | TriggerDescription$ When you do, CARDNAME becomes a copy of another target creature you control, except it has this ability and "Whenever this creature deals combat damage to a player, you get that many {E}."
+SVar:TrigCopy:DB$ Clone | AddTriggers$ TrigDamage | ValidTgts$ Creature.Other+YouCtrl | TgtPrompt$ Select another target creature you control to copy | AddSVars$ TrigDamage,TrigEnergy
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/aurora_shifter.txt:9]: exception incomplete

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/aurora_shifter.txt:9]: I've used `GainThisAbility$ True` but this caused a crash

File: forge-gui/res/cardsfolder/upcoming/march_from_velis_vel.txt [hunk 10]
```diff
@@ -0,0 +1,7 @@
+Name:March from Velis Vel
+ManaCost:2 U
+Types:Sorcery
+A:SP$ ChooseType | Defined$ You | Type$ Nonbasic Land | SubAbility$ DBCloneAll | SpellDescription$ Choose a nonbasic land type. Each land you control of that type becomes a copy of target creature you control until end of turn and gains haste until end of turn.
+SVar:DBCloneAll:DB$ Clone | Defined$ Targeted | CloneTarget$ Valid Land.YouCtrl+ChosenType | AddKeywords$ Haste | Duration$ UntilEndOfTurn | Duration$ UntilEndOfTurn | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Choose target creature you control
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/march_from_velis_vel.txt:5]: duration twice

File: forge-gui/res/cardsfolder/upcoming/tempt_with_mayhem.txt [hunk 11]
```diff
@@ -0,0 +1,9 @@
+Name:Tempt with Mayhem
+ManaCost:1 R R
+Types:Instant
+A:SP$ RepeatEach | ValidTgts$ Instant,Sorcery | TgtZone$ Stack | TgtPrompt$ Select target instant or sorcery spell | RepeatSubAbility$ DBCopy | RepeatPlayers$ Player.Opponent | SubAbility$ DBCopySelf | RepeatOptionalForEachPlayer$ True | RememberDeciders$ True | RepeatOptionalMessage$ Do you want to copy the targeted spell? | SpellDescription$ Tempting offer — Choose target instant or sorcery spell. Each opponent may copy that spell and may choose new targets for the copy they control. You copy that spell once plus an additional time for each opponent who copied the spell this way. You may choose new targets for the copies you control.
+SVar:DBCopy:DB$ CopySpellAbility | Defined$ Targeted | Amount$ 1 | MayChooseTarget$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/tempt_with_mayhem.txt:5]: who should control these copies...?

File: forge-game/src/main/java/forge/game/ability/AbilityUtils.java [hunk 12]
```diff
@@ -1997,7 +1997,7 @@ public static int xCount(Card c, final String s, final CardTraitBase ctb) {
             return doXMath(calculateAmount(c, sq[!isUnlinkedFromCastSA(ctb, c) && c.getKickerMagnitude() > 0 ? 1 : 2], ctb), expr, c, ctb);
         }
         if (sq[0].startsWith("Escaped")) {
-            return doXMath(calculateAmount(c, sq[!isUnlinkedFromCastSA(ctb, c) && c.getCastSA() != null && c.getCastSA().isEscape() ? 1 : 2], ctb), expr, c, ctb);
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/ability/AbilityUtils.java:2000]: Escaped should be linked the to Escape Cost?

Comment (tool4ever) [reply forge-game/src/main/java/forge/game/ability/AbilityUtils.java:2000]: I don't think so:
> 702.138b. A spell or permanent "escaped" if that spell or the spell that became that permanent as it resolved was cast from a graveyard with an escape ability.

Rulings on relevant cards agree with that:
>  Uro’s first ability causes you to sacrifice it if you didn’t cast it, or if it was cast using any permission other than an escape ability.

File: forge-gui/res/cardsfolder/upcoming/sunken_palace.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/sunken_palace.txt:7]: You're right it did work. It's just hard to use the "special" mana in testing

**Review by tool4ever — APPROVED** [review]

**Pull request merged**